src/components/CreateHabit.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import { StyleSheet } from 'react-native';
+import { Button, Dialog, TextInput, Portal, Surface } from 'react-native-paper';
+//REDUX
+import { connect } from 'react-redux';
+import { createHabit } from '../redux/actions';
+
+const styles = StyleSheet.create({
+    dialog: {
+        height: 400,
+        width: 325,
+    },
+    surface: {
+        padding: 8,
+        margin: 8,
+        marginLeft: 25,
+        height: 80,
+        width: 325,
+        alignItems: 'center',
+        justifyContent: 'center',
+        elevation: 4,
+    },
+});
+
+export class CreateHabit extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { visible: false, tempTitle: '', tempLimit: '' };
+        this.showDialog = this.showDialog.bind(this);
+        this.hideDialog = this.hideDialog.bind(this);
+        this.createHabit = this.createHabit.bind(this);
+        this.setTempTitle = this.setTempTitle.bind(this);
+        this.setTempLimit = this.setTempLimit.bind(this);
+    }
+
+    showDialog = () => this.setState({ visible: true });
+
+    hideDialog = () => this.setState({ visible: false });
+
+    setTempTitle = (title) => { this.setState({ tempTitle: title }) };
+
+    setTempLimit = (limit) => { this.setState({ tempLimit: limit }) };
+
+    createHabit = () => {
+        this.props.createHabit(this.state.tempTitle, this.state.tempLimit);
+        this.hideDialog();
+    }
```

Comment:
You could avoid repeating a lot of boilerplate stuff here with hooks:

```js
export const CreateHabit = () => {
  const [showDialoag, setShowDialog] = useState(false)

  return (
    <Button onPress={() => this.setShowDialog(true)}> Create new habit</Button>
  )
}
```

- [React Hooks](https://reactjs.org/docs/hooks-intro.html)
- [Redux Hooks](https://react-redux.js.org/api/hooks)

Reply:
Refactored component using hooks